crates/astria-composer/src/searcher/mod.rs

```diff
@@ -318,3 +324,107 @@ impl Searcher {
         Ok(())
     }
 }
+
+fn reconnect_exited_collector(
+    collector_statuses: &mut HashMap<String, watch::Receiver<collector::Status>>,
+    collector_tasks: &mut JoinMap<String, eyre::Result<()>>,
+    new_transactions_tx: Sender<collector::Transaction>,
+    rollups: &HashMap<String, String>,
+    rollup: String,
+    exit_result: Result<eyre::Result<()>, JoinError>,
+) {
+    match exit_result {
+        Ok(Ok(())) => warn!("collector exited unexpectedly; reconnecting"),
+        Ok(Err(e)) => {
+            let error: &(dyn std::error::Error + 'static) = e.as_ref();
+            warn!(error, "collector exit with error; reconnecting");
+        }
+        Err(e) => {
+            let error = &e as &(dyn std::error::Error + 'static);
+            warn!(error, "collector task failed; reconnecting");
+        }
+    }
+    let Some(url) = rollups.get(&rollup) else {
+        warn!(
+            "rollup should have had an entry in the rollup->url map but doesn't; not reconnecting \
+             it"
+        );
```

Comment:
should never happen right?

Reply:
Yeah, it shouldn't. I bumped it to `error!`.